src/pages/BestVendorsPage/BestVendorsList/index.tsx

```diff
@@ -10,16 +11,90 @@ import {
   PaddingVariants,
   StyleVariants,
 } from 'src/components/shared/StyledButton/types';
-import { useGetBestVendorsQuery } from 'src/redux/bestVendors/bestVendorsService';
+import useToast from 'src/components/shared/toasts/components/ToastProvider/ToastProviderHooks';
+import {
+  useFollowVendorMutation,
+  useGetBestVendorsQuery,
+  useUnfollowVendorMutation,
+} from 'src/redux/bestVendors/bestVendorsService';
 import { IProductFilters } from 'src/redux/product/types';
 
+interface FollowStatus {
+  [vendorId: string]: boolean;
+}
+
 type BestVerdorsListProps = {
   filters: IProductFilters;
 };
 
 function BestVendorsList({ filters }: BestVerdorsListProps) {
   const { t } = useTranslation();
+  const { showToast } = useToast();
   const { data: bestVendors } = useGetBestVendorsQuery(filters);
+  const [followVendor] = useFollowVendorMutation();
+  const [unFollowVendor] = useUnfollowVendorMutation();
+
+  const [followStatus, setFollowStatus] = useState<FollowStatus>(() => {
+    const storedStatus = localStorage.getItem('followStatus');
+
+    return storedStatus ? JSON.parse(storedStatus) : {};
+  });
+
+  useEffect(() => {
+    if (bestVendors) {
+      setFollowStatus((prevStatus) => {
+        const updatedStatus = bestVendors.reduce((acc, vendor) => {
+          acc[vendor.vendorId] = prevStatus[vendor.vendorId] ?? false;
+
+          return acc;
+        }, {} as FollowStatus);
+
+        return updatedStatus;
+      });
+    }
+  }, [bestVendors]);
+
+  useEffect(() => {
+    localStorage.setItem('followStatus', JSON.stringify(followStatus));
+  }, [followStatus]);
```

Comment:
nonon why we stire it in localStorage. 1) I dont understand why it needed 2) redux persist resist for such things

Reply:
fixed here https://github.com/ZenBit-Tech/code-lions_fe/pull/78/commits/3a39c8eb3d4cb874b7ba51b29cece6f903159f07 it is needed to show the user whether  he has already followed or not certain vendor in case of page reloading or leaving the page